### PR TITLE
More calculation methods

### PR DESCRIPTION
**Pull request by mirfatif**

### CODE REVIEW

**Review by ahmedre — COMMENTED** [review]
i'd like to keep this list matching the list in the Swift Adhan code base - many of these (Jafari, Gulf, France, Russia) aren't in the Swift library so don't want to add them right now.

File: adhan/src/main/java/com/batoulapps/adhan/CalculationMethod.java [hunk 1]
```diff
@@ -66,6 +66,42 @@
...
+   * Shia Ithna Ashari, Leva Institute, Qum
+   * Uses Fajr angle of 16, Maghrib angle of 4 and Isha angle of 14
+   */
+  JAFARI,
+
+  /**
+   * Shia, Institute of Geophysics, University of Tehran
+   * Uses Fajr angle of 17.7, Maghrib angle of 4.5 and Isha angle of 14
+   */
+  TEHRAN,
```
Comment (ahmedre) [inline adhan/src/main/java/com/batoulapps/adhan/CalculationMethod.java:79]: let's match the documentation of the swift library -

```
Institute of Geophysics, University of Tehran. Early Isha time with an angle of 14°. Slightly later Fajr time with an angle of 17.7°.
  Calculates Maghrib based on the sun reaching an angle of 4.5° below the horizon.
```

File: adhan/src/main/java/com/batoulapps/adhan/HighLatitudeRule.java [hunk 2]
```diff
@@ -4,6 +4,8 @@
  * Rules for dealing with Fajr and Isha at places with high latitudes
  */
 public enum HighLatitudeRule {
+  NONE,
```
Comment (ahmedre) [inline adhan/src/main/java/com/batoulapps/adhan/HighLatitudeRule.java:7]: let's remove to remain consistent with iOS

File: adhan/src/main/java/com/batoulapps/adhan/CalculationParameters.java [hunk 3]
```diff
@@ -113,6 +143,9 @@ private NightPortions(double fajr, double isha) {
 
   NightPortions nightPortions() {
     switch (this.highLatitudeRule) {
+      case NONE: {
+        return null;
+      }
```
Comment (ahmedre) [inline adhan/src/main/java/com/batoulapps/adhan/CalculationParameters.java:148]: same here

File: adhan/src/main/java/com/batoulapps/adhan/PrayerTimes.java [hunk 4]
```diff
@@ -94,17 +104,17 @@ public PrayerTimes(Coordinates coordinates, DateComponents date, CalculationPara
...
         double portion = nightPortions.fajr;
         long nightFraction = (long) (portion * night / 1000);
         safeFajr = CalendarUtil.add(
             sunriseComponents, -1 * (int) nightFraction, Calendar.SECOND);
       }
 
-      if (tempFajr == null || tempFajr.before(safeFajr)) {
+      if (safeFajr != null && (tempFajr == null || tempFajr.before(safeFajr))) {
         tempFajr = safeFajr;
       }
```
Comment (ahmedre) [inline adhan/src/main/java/com/batoulapps/adhan/PrayerTimes.java:119]: with the above changes we can undo these

File: adhan/src/main/java/com/batoulapps/adhan/PrayerTimes.java [hunk 5]
```diff
@@ -124,17 +134,17 @@ public PrayerTimes(Coordinates coordinates, DateComponents date, CalculationPara
...
+        } else if (nightPortions != null) {
           double portion = nightPortions.isha;
           long nightFraction = (long) (portion * night / 1000);
           safeIsha = CalendarUtil.add(sunsetComponents, (int) nightFraction, Calendar.SECOND);
         }
 
-        if (tempIsha == null || (tempIsha.after(safeIsha))) {
+        if (safeIsha != null && (tempIsha == null || (tempIsha.after(safeIsha)))) {
           tempIsha = safeIsha;
         }
```
Comment (ahmedre) [inline adhan/src/main/java/com/batoulapps/adhan/PrayerTimes.java:149]: and these

**Comment by z3bi** [comment]
Closing stale PR

**Pull request closed without merging**